presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java

```diff
@@ -196,6 +207,71 @@ public void testCacheRead()
         assertEquals(getCacheStats().getRemoteReads(), intermittentRemoteReads);
     }
 
+    @Test
+    public void testLargeFile()
+            throws Exception
+    {
+        byte[] randomData = new byte[(int) LARGE_FILE_SIZE.toBytes()];
+        new Random().nextBytes(randomData);
+
+        Path file = getStoragePath("large_file");
+        try (FSDataOutputStream outputStream = nonCachingFileSystem.create(file)) {
+            outputStream.write(randomData);
+        }
+
+        long beforeRemoteReads = getCacheStats().getRemoteReads();
+        long beforeCachedReads = getCacheStats().getCachedReads();
+
+        assertTrue(Arrays.equals(randomData, readFileBytes(cachingFileSystem.open(file))));
+
+        // stats are propagated asynchronously, wait for them
+        sleep(1000L);
+        long intermittentRemoteReads = getCacheStats().getRemoteReads();
+        assertGreaterThan(intermittentRemoteReads, beforeRemoteReads);
+
+        assertTrue(Arrays.equals(randomData, readFileBytes(cachingFileSystem.open(file))));
+
+        // stats are propagated asynchronously, wait for them
+        sleep(1000L);
+        long afterCachedReads = getCacheStats().getCachedReads();
+        long afterRemoteReads = getCacheStats().getRemoteReads();
+        assertGreaterThan(afterCachedReads, beforeCachedReads);
```

Comment:
`intermittent` and `after` name prefixes are poor. Use something more explicit. `firstReadCachedRead`, `secondReadCachedReads`?